crates/client/block_import/src/verify_apply/contracts.rs

```diff
@@ -129,3 +129,85 @@ fn contract_state_leaf_hash(
     // computes the contract state leaf hash
     Ok(Pedersen::hash(&Pedersen::hash(&Pedersen::hash(&class_hash, &storage_root), &nonce), &Felt::ZERO))
 }
+
+#[cfg(test)]
+mod contract_trie_root_tests {
+    use super::*;
+    use mp_chain_config::ChainConfig;
+    use std::sync::Arc;
+
+    fn setup_test_backend() -> Arc<MadaraBackend> {
+        let chain_config = Arc::new(ChainConfig::test_config());
+        MadaraBackend::open_for_testing(chain_config)
+    }
+
+    #[test]
+    fn test_contract_trie_root_success() {
+        let backend = setup_test_backend();
+
+        // Create dummy data
+        let deployed_contracts = vec![DeployedContractItem {
+            address: Felt::from_hex_unchecked("0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef"),
+            class_hash: Felt::from_hex_unchecked("0xfedcba0987654321fedcba0987654321fedcba0987654321fedcba0987654321"),
+        }];
+
+        let replaced_classes = vec![ReplacedClassItem {
```

Comment:
I don’t think this case is possible because the contract is not deployed yet. I don’t believe we can use replaced_classes within a single genesis block

Reply:
here we are doing the functional testing only, hence using dummy values. and I agree this case might not exist in real.